sdl-core/src/main/scala/io/smartdatalake/workflow/dataobject/ExpectationValidation.scala

```diff
@@ -55,48 +59,41 @@ private[smartdatalake] trait ExpectationValidation { this: DataObject with Smart
    */
   def expectations: Seq[Expectation]
 
-  def setupConstraintsAndJobExpectations(df: GenericDataFrame)(implicit context: ActionPipelineContext): (GenericDataFrame, DataFrameObservation) = {
+  def setupConstraintsAndJobExpectations(df: GenericDataFrame, defaultExpectationsOnly: Boolean = false, predicateTolerant: Boolean = false, additionalJobAggExpressionColumns: Seq[GenericColumn] = Seq())(implicit context: ActionPipelineContext): (GenericDataFrame, DataFrameObservation) = {
     // add constraint validation column
-    val dfConstraints = setupConstraintsValidation(df)
+    val dfConstraints = if (defaultExpectationsOnly) df else setupConstraintsValidation(df)
     // setup job expectations as DataFrame observation
     val jobExpectations = expectations.filter(_.scope == ExpectationScope.Job)
     val (dfJobExpectations, observation) = {
       implicit val functions: DataFrameFunctions = DataFrameSubFeed.getFunctions(df.subFeedType)
-      val expectationColumns = (defaultExpectations ++ jobExpectations).flatMap(_.getAggExpressionColumns(this.id))
-      setupObservation(dfConstraints, expectationColumns, context.isExecPhase)
+      val expectationColumns = (defaultExpectations ++ (if (defaultExpectationsOnly) Seq() else jobExpectations)).flatMap(_.getAggExpressionColumns(this.id)) ++ additionalJobAggExpressionColumns
```

Comment:
how can there be jobExpectations if the user defined no expectation? I would assume jobExpectations is empty anyway if the user did not specify any expectations